### DISCUSSION TITLE
Fitipower FC0012 tuner Segmentation Fault

### DISCUSSION THREAD

**Discussion by MarkYoungIW** [jvde-github/AIS-catcher#316] [General]
Hi, I have set up a RPi Zero WH with a clean installation of Raspberry Pi OS Lite (32-Bit) and have installed AIS Catcher and edited the config.cmd file appropriately.

For some reason though I keep coming to the same fault.
 
```
pi@raspberrypi:~ $ AIS-catcher -v 10
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Segmentation fault
pi@raspberrypi:~ $
```

```
pi@raspberrypi:~ $ sudo systemctl status ais-catcher.service
● ais-catcher.service - AIS-catcher Service
     Loaded: loaded (/etc/systemd/system/ais-catcher.service; enabled; preset: >
     Active: active (running) since Wed 2024-10-02 12:31:05 BST; 17min ago
   Main PID: 1017 (AIS-catcher)
      Tasks: 5 (limit: 387)
        CPU: 17min 33.076s
     CGroup: /system.slice/ais-catcher.service
             └─1017 /usr/bin/AIS-catcher -C /etc/AIS-catcher/config.json -d:0 ->

Oct 02 12:48:42 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:44 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:45 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:47 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:48 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:50 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:51 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:53 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:54 raspberrypi bash[1017]: RTLSDR: timeout.
Oct 02 12:48:56 raspberrypi bash[1017]: RTLSDR: timeout.
lines 1-19/19 (END)
```

```
pi@raspberrypi:~ $ AIS-catcher -l
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Found 1 device(s):
0: Realtek, RTL2838UHIDIR, SN: 00000001
pi@raspberrypi:~ $ 
```

I would be grateful for any advice to help me get this running.

Many Thanks

**Comment by MarkYoungIW** [comment]
I am suspecting now that the RPi Zero is not feeding the dongle with sufficient power.

My reasoning for this is because I just took the card out of the zero, put it in a RPi 3B, and everything is working perfectly.

**Reply by jvde-github** [reply]
Are you running AIS-catcher -v  at the same time the systemctl process is running?  The systemctl status suggests that something is running but no data is flowing. 

If so, what do you see if the service is stopped: sudo systemctl stop ais-catcher and then run the program AIS-catcher -v 

It could be a power issue but interesting that the program crashes with a segmentation fault. Do you get the same with the native rtl-sdr utilities like rtl_test and rtl_sdr?

if the dataflow is too high for the device might try to run on a lower sampling rate: -s 288K for example

**Reply by MarkYoungIW** [reply]
I have to be honest, where the RPi3 is working fine, I have switched to that now, but the zero still does this.

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher
pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Device    : RTL2838UHIDIR
Settings  : rate 1536K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count 24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON fp_ds OFF dsk OFF
Segmentation fault
pi@raspberrypi:~ $


Or this:

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher
pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Segmentation fault

**Reply by MarkYoungIW** [reply]
The config file is set like this, so I am unsure why it say's otherwise.


#Fast Downsampling Switch On
-F

#Fast Downsampling Rate
-s 288K

**Reply by jvde-github** [reply]
Settings look good for the Zero (run at lower sample rate to reduce data throughput for the Zero). There is some sort of issue but don't know what it is. It can be in the rtl-sdr and libusb driver but it is difficult to replicate. Is it worthwhile to investigate more or are you sticking with the 3?

If I have some free time I will try to see if I can replicate on a zero. Can you share which dongle you are precisely using?

Thanks, J

**Reply by MarkYoungIW** [reply]
The above issues I have been having are with the black dongle in the photo. The silver one also does the same as below.

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher
pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Rafael Micro R820T tuner
Segmentation fault
pi@raspberrypi:~ $

![IMG_20241031_120846](https://github.com/user-attachments/assets/c8e49e1e-44c4-47ac-892e-5385a4c67820)

**Reply by MarkYoungIW** [reply]
Actually, quickly to stray away from this, can you filter out an MMSI in the software? I have one MMSI in range that appears to be fake, and it messes up the screen pretty quickly on day's it is active.

Thanks

**Reply by jvde-github** [reply]
Added something to the program for that just now. If you want to filter out an mmsi of the webviewer:

AIS-catcher -N 8100 filter ON block_mmsi 23435235

**Reply by jvde-github** [reply]
Weird that it does not wrong with your silver dongle. Either something with your rtl-sdr driver or an issue with the Zero Pi (WH). Anyone else having an issue on the Pi zero WH?

**Reply by MarkYoungIW** [reply]
I will spice up a wire later and try feeding the dongles with power bypassing the Pi, just incase it is that.

I do have another zero, but it isn't wireless unfortunately.

I will reinstall both Raspberry Pi OS Lite (32-Bit) and AIS Catcher from scratch again and see if it is still the same.

While I am happy using the RPi3, it would be good to work out why this issue has occured for me.

Does the '-N 8100 filter ON block_mmsi 23435235' also block the sentance from being forwarded on by AIS catcher?

Thanks.

**Comment by MarkYoungIW** [comment]
OK, I have spliced up a cable so that the dongle is being fed power seperately and not relying on the Zero, and I have this now.

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher
pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Rafael Micro R820T tuner
Device    : RTL2838UHIDIR
Settings  : rate 1536K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count                                                                              24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON                                                                              fp_ds OFF dsk OFF
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
----------------------
[AIS engine v0.60 #0-0]                 total: 0 msgs
Reattaching kernel driver failed!
pi@raspberrypi:~ $

**Comment by MarkYoungIW** [comment]
Actually, I think that was a one off, as it has gone back to this instead now.

pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Rafael Micro R820T tuner
Segmentation fault
pi@raspberrypi:~ $

**Reply by jvde-github** [reply]
CB5 is device disconnected. Will give it tomorrow a try with a zero WH as well and see what it says.

**Reply by MarkYoungIW** [reply]
In the meantime, I will set up the SSD from scratch.

**Reply by jvde-github** [reply]
can you then install the program via the script on the github page, so we run the same set up?

**Reply by MarkYoungIW** [reply]
Yes, no problems. It was installed via the script the first time around.

**Reply by MarkYoungIW** [reply]
OK, I have started over from scratch. Clean install of Raspberry Pi OS Lite (32-Bit).

Installed AIS Catcher using:-

sudo bash -c "$(wget -qO- https://raw.githubusercontent.com/jvde-github/AIS-catcher/main/scripts/aiscatcher-install)"

Plugged the dongle in and:-

pi@raspberrypi:~ $ /usr/bin/AIS-catcher -v
AIS-catcher (build Oct 31 2024) v0.60-200-ge693b3f6
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Device    : RTL2838UHIDIR
Settings  : rate 1536K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count                                                                              24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON                                                                              fp_ds OFF dsk OFF
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
cb transfer status: 5, canceling...
[ERROR] RTLSDR: lost device. Terminating.
----------------------
[AIS engine v0.60 #0-0]                 total: 0 msgs
Reattaching kernel driver failed!

Or

pi@raspberrypi:~ $ /usr/bin/AIS-catcher -v
AIS-catcher (build Oct 31 2024) v0.60-200-ge693b3f6
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Segmentation fault
pi@raspberrypi:~ $

**Comment by MarkYoungIW** [comment]
And the black dongle is doing the same as before, or occasionally this, so I think we can rule out a power issue.

pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Oct  2 2024) v0.60-127-g3cd6272b
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Device    : RTL2838UHIDIR
Settings  : rate 1536K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count                                                                              24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON                                                                              fp_ds OFF dsk OFF
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.
RTLSDR: timeout.
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0                                                                              msg/s
RTLSDR: timeout.

**Reply by jvde-github** [reply]
I tried yesterday raspberry pi zero rev 1.1 WH with OS 32 lite and 3 different dongles. All worked without the issues you have seen. I did not try with the silver V3 (tested with orange flightaware, mini dongle and V4). Will test a bit more but is difficult to replicate.

Are you using an older Zero?

**Comment by jvde-github** [comment]
Other question, are utilities like rtl_test working normally?
They should be installed already

**Reply by MarkYoungIW** [reply]
rtl_test appears to be working correctly. Pi Zero is v1.1 as pictured.

I will try and grab another zero in the next few days as I have a broken one I need to replace now anyway, and try it with that. Try and narrow down if it's perhaps the dongles or not.

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher.service
pi@raspberrypi:~ $ rtl_test
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Detached kernel driver
Found Fitipower FC0012 tuner
Supported gain values (5): -9.9 -4.0 7.1 17.9 19.2
Sampling at 2048000 S/s.

Info: This tool will continuously read from the device, and report if
samples get lost. If you observe no further output, everything is fine.

Reading samples in async mode...


![IMG_20241102_130507](https://github.com/user-attachments/assets/5a8709ad-1241-4e23-b2af-82b2f8112e91)

**Reply by jvde-github** [reply]
This is precisely the same as I have been testing with last two days. Also installed OS 32bit Lite and same install script. 

You see the same crash with?
```
rtl_sdr -f 162000000 -s 288000  - | AIS-catcher -r cu8 . -s 288000 -v
```
If it happens here as well it is something in AIS-catcher...

**Reply by MarkYoungIW** [reply]
Results as follows:

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher.service
pi@raspberrypi:~ $ rtl_sdr -f 162000000 -s 288000  - | AIS-catcher -r cu8 . -s 288000 -v
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Detached kernel driver
AIS-catcher (build Nov  3 2024) v0.60-230-g7bdc62d1
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Device    : File (RAW)
Settings  : rate 288K format CU8 file . loop OFF
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON fp_ds OFF dsk OFF
Found Fitipower FC0012 tuner
Sampling at 288000 S/s.
Tuned to 162000000 Hz.
Tuner gain set to automatic.
Reading samples in async mode...
----------------------
[AIS engine v0.60 #0-0]                 total: 0 msgs
pi@raspberrypi:~ $

**Reply by MarkYoungIW** [reply]
I am getting a different result from a cold start up vs stopping the service and restarting it.

Sometimes I get this from a cold start:

pi@raspberrypi:~ $ sudo systemctl status ais-catcher.service
● ais-catcher.service - AIS-catcher Service
Loaded: loaded (/etc/systemd/system/ais-catcher.service; enabled; preset: >
Active: activating (auto-restart) (Result: signal) since Sun 2024-11-03 16>
Process: 1030 ExecStart=/bin/bash -c /usr/bin/AIS-catcher -C /etc/AIS-catch>
" " ") (code=killed, signal=SEGV)
Main PID: 1030 (code=killed, signal=SEGV)
CPU: 508ms

Nov 03 16:29:36 raspberrypi systemd[1]: ais-catcher.service: Main process exite>
Nov 03 16:29:36 raspberrypi systemd[1]: ais-catcher.service: Failed with result>
lines 1-10/10 (END)...skipping...
● ais-catcher.service - AIS-catcher Service
Loaded: loaded (/etc/systemd/system/ais-catcher.service; enabled; preset: enabled)
Active: activating (auto-restart) (Result: signal) since Sun 2024-11-03 16:29:36 GMT; 3s ago
Process: 1030 ExecStart=/bin/bash -c /usr/bin/AIS-catcher -C /etc/AIS-catcher/config.json $(/bin/grep -v "^#" /etc/AIS-catcher/config.cmd | /usr/bin/tr "
" " ") (code=killed, signal=SEGV)
Main PID: 1030 (code=killed, signal=SEGV)
CPU: 508ms

Nov 03 16:29:36 raspberrypi systemd[1]: ais-catcher.service: Main process exited, code=killed, status=11/SEGV
Nov 03 16:29:36 raspberrypi systemd[1]: ais-catcher.service: Failed with result 'signal'.
~
~
~
~
~
~
~

But then other times the cold start brings up the web interface as soon as the RPi comes up, and the service status is as below. I have then continued to stop the service, ais-catcher -v, and restart the service. After the service is restarted, the web interface does not come up. Stopping the service does seem to take a very long time.

pi@raspberrypi:~ $ sudo systemctl status ais-catcher.service
● ais-catcher.service - AIS-catcher Service
Loaded: loaded (/etc/systemd/system/ais-catcher.service; enabled; preset: >
Active: active (running) since Sun 2024-11-03 16:30:30 GMT; 1min 30s ago
Main PID: 535 (AIS-catcher)
Tasks: 5 (limit: 387)
CPU: 39.125s
CGroup: /system.slice/ais-catcher.service
└─535 /usr/bin/AIS-catcher -C /etc/AIS-catcher/config.json -d:0 -N>

Nov 03 16:31:46 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:48 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:50 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:51 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:53 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:54 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:56 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:57 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:59 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:32:00 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
lines 1-19/19 (END)...skipping...
● ais-catcher.service - AIS-catcher Service
Loaded: loaded (/etc/systemd/system/ais-catcher.service; enabled; preset: enabled)
Active: active (running) since Sun 2024-11-03 16:30:30 GMT; 1min 30s ago
Main PID: 535 (AIS-catcher)
Tasks: 5 (limit: 387)
CPU: 39.125s
CGroup: /system.slice/ais-catcher.service
└─535 /usr/bin/AIS-catcher -C /etc/AIS-catcher/config.json -d:0 -N 8100 CDN /etc/AIS-catcher/webassets LAT 50.7570>

Nov 03 16:31:46 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:48 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:50 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:51 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:53 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:54 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:56 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:57 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:31:59 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
Nov 03 16:32:00 raspberrypi bash[535]: [ERROR] RTLSDR: timeout.
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher.service
pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Nov 3 2024) v0.60-230-g7bdc62d1
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Fitipower FC0012 tuner
Device : RTL2838UHIDIR
Settings : rate 1536K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count 24
Segmentation fault
pi@raspberrypi:~ $ sudo systemctl start ais-catcher.service
pi@raspberrypi:~ $ sudo systemctl status ais-catcher.service
● ais-catcher.service - AIS-catcher Service
Loaded: loaded (/etc/systemd/system/ais-catcher.service; enabled; preset: enabled)
Active: activating (auto-restart) (Result: signal) since Sun 2024-11-03 16:34:32 GMT; 5s ago
Process: 849 ExecStart=/bin/bash -c /usr/bin/AIS-catcher -C /etc/AIS-catcher/config.json $(/bin/grep -v "^#" /etc/AIS-catcher/config.cmd | /usr/bin/tr "
" " ") (code=killed, signal=SEGV)
Main PID: 849 (code=killed, signal=SEGV)
CPU: 490ms

Nov 03 16:34:32 raspberrypi systemd[1]: ais-catcher.service: Failed with result 'signal'.
pi@raspberrypi:~ $

**Reply by MarkYoungIW** [reply]
The MMSI filter works great on the web interface, is there any way that this could be expanded so it doesn't forward the dropped MMSI data on via UDP/TCP share too?

Thanks

**Reply by jvde-github** [reply]
yes, you can add a filter to any output channel, e.g. ```AIS-catcher -u 127.0.0.1 5012 filter ON block_mmsi 23435235 -S 5012 filter ON block_mmsi 23435235 -N 8100 filter ON block_mmsi 23435235```

**Reply by MarkYoungIW** [reply]
Thank you. Filtering work's perfectly and removed my problem MMSI.

**Comment by jvde-github** [comment]
Made a small additional to the program to give more insight into why we have the segmentation fault.  Can you install again AIS-catcher on the troubled RPI Zero with debug symbols:
```
sudo bash -c "$(wget -qO- https://raw.githubusercontent.com/jvde-github/AIS-catcher/main/scripts/aiscatcher-install)" _ -d
```` 

Note the `_ -d` at the end. This is a suboptimal version just for debugging so not really for production.

And then run the program as you did, ./AIS-catcher -v and share the output if you have the segmentation fault again. Hopefully provides some guidance where this happens in the program.

**Reply by MarkYoungIW** [reply]
Results now are:

pi@raspberrypi:~ $ sudo systemctl stop ais-catcher.service
pi@raspberrypi:~ $ AIS-catcher -v
AIS-catcher (build Nov  4 2024) v0.60-232-g0f72aab6
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
**** Creating stat and msg_count
**** Setting up receiver: 0
Detached kernel driver
Found Fitipower FC0012 tuner
**** Starting output devices
**** Starting servers
**** Starting statistics
**** Starting receivers

AIS-catcher - Internal error: signal 11 - Segmentation fault

Stack trace:
[1]: /lib/arm-linux-gnueabihf/libc.so.6: __default_sa_restorer +0 (/lib/arm-linu                                                                             x-gnueabihf/libc.so.6)
Segmentation fault
pi@raspberrypi:~ $


Thanks

**Reply by jvde-github** [reply]
Thanks! That is quick. Unfortunately, not a lot of information where the issue occurs. Need to think a bit more about this. Not very obvious.

**Reply by MarkYoungIW** [reply]
No problems. Always happy to do some testing if needed.

Thanks again.
